Fix linked-on-or-after check for using process assertions off the main thread

bnham opened the pull request:
#### 838b2391f4ce109df730be706b430f5cb11a8146
<pre>
Fix linked-on-or-after check for using process assertions off the main thread
<a href="https://bugs.webkit.org/show_bug.cgi?id=302758">https://bugs.webkit.org/show_bug.cgi?id=302758</a>
<a href="https://rdar.apple.com/164866073">rdar://164866073</a>

Reviewed by Chris Dumez.

In 302397@main, I passed in the wrong SDKAlignedBehavior for the check on whether to fault or assert
when manipulating process assertions off the main thread.

This fixes that issue, and also renames the symbol to make it clearer that the crash is due to an
error in the application rather than an issue in WebKit. The existing assert with message macro
doesn&apos;t seem to always result in a message in the crash log from what I&apos;ve seen.

* Source/WebKit/UIProcess/ProcessThrottler.cpp:
(WebKit::crashDueToApplicationCallingMainThreadOnlyWebKitAPIFromBackgroundThread):
(WebKit::assertIfCalledFromBackgroundThread):

Canonical link: <a href="https://commits.webkit.org/303266@main">https://commits.webkit.org/303266@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d7149f036396d6bf338ce8b5fb90687ae8f853b4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131776 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4268 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42780 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139285 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83649 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ca176372-008a-4b24-8c0f-8a0897b99f62) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4198 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4030 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100721 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/23789fe5-6c31-4b06-82d5-997132f3cad2) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134722 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/3016 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118023 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81505 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a51c20ba-6e19-4e88-8e86-d59a0bc58027) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2903 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/750 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82507 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111637 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141931 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3937 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36727 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109092 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4018 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3460 "Found 2 new test failures: imported/w3c/web-platform-tests/fullscreen/rendering/ua-style-iframe.html imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue_function/not_root_selector.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109255 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27691 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2992 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114304 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57147 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3991 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32712 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3823 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67438 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4083 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3951 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/ProcessThrottler.cpp:
```diff
@@ -174,6 +171,14 @@ static void assertIfCalledFromBackgroundThread()
 
+static void assertIfCalledFromBackgroundThread()
+{
+    if (isMainRunLoop())
```
We may want to `[[likely]]` this branch.

bnham committed:
Fix linked-on-or-after check for using process assertions off the main thread
https://bugs.webkit.org/show_bug.cgi?id=302758
rdar://164866073

Reviewed by Chris Dumez.

In 302397@main, I passed in the wrong SDKAlignedBehavior for the check on whether to fault or assert
when manipulating process assertions off the main thread.

This fixes that issue, and also renames the symbol to make it clearer that the crash is due to an
error in the application rather than an issue in WebKit. The existing assert with message macro
doesn't seem to always result in a message in the crash log from what I've seen.

* Source/WebKit/UIProcess/ProcessThrottler.cpp:
(WebKit::crashDueToApplicationCallingMainThreadOnlyWebKitAPIFromBackgroundThread):
(WebKit::assertIfCalledFromBackgroundThread):

Canonical link: https://commits.webkit.org/303266@main
